back/engines/commercial/custom_maps/app/models/custom_maps/map_config.rb

```diff
@@ -49,11 +49,9 @@ def center_geojson=(geojson)
     end
 
     def project_id
-      resource.project_id if resource_type == 'Project'
-    end
+      return nil unless mappable_type == 'Project'
 
-    def custom_field_id
-      resource.custom_field_id if resource_type == 'CustomField'
+      mappable&.id
```

Comment:
The previous, erroneous version, was not used until the new `SideFxMapConfigService` was introduced, since this method is called by `LogActivityJob#initialize`

I also removed the redundant (unused) `custom_field_id` method, which also would have erorred if ever called.